crates/telemetry/src/config.rs

```diff
@@ -0,0 +1,72 @@
+use envconfig::Envconfig;
+use tracing::Level;
+
+/// Provides configuration for the telemetry system.
+///
+/// Largely consists of OpenTelemetry variables defined
+/// [here](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration)
+/// but also has some Spin specific configuration.
+#[derive(Debug)]
+pub struct Config {
+    /// Disables OpenTelemetry for all signals (traces and metrics).
+    pub otel_sdk_disabled: bool,
+
+    /// Determines the verbosity of the OpenTelemetry tracing layer.
+    ///
+    /// This is a Spin specific value in the sense that it is not defined by the OpenTelemetry
+    /// standard. This is what allows us to have different levels of verbosity in the fmt layer and
+    /// the otel layer. The fmt layer respects `RUST_LOG` and the otel layer respects
+    /// `SPIN_OTEL_TRACING_LEVEL`.
+    ///
+    /// Note: Currently SPIN_OTEL_TRACING_LEVEL defaults to info and makes no distinction between
+    /// spans from Spin crates or external dependencies. In the future we may desire to provide a
+    /// more curated experience of what spans show by default. This curated experience could be
+    /// created by annotating every Spin span with a special attribute that can be filtered on. Then
+    /// if the user opts to set SPIN_OTEL_TRACING_LEVEL this curated filter would be turned off and
+    /// we would allow spans from any library at the level defined by the user.
```

Comment:
I believe we could also use the same fmt filtering that `RUST_LOG` uses: https://docs.rs/tracing-subscriber/0.3.18/tracing_subscriber/filter/struct.EnvFilter.html

Reply:
We could certainly use the EnvFilter if we wanted. I opted for the LevelFilter b/c I wanted to expose less knobs to the end user. But, my whole note there is basically saying "hey eventually we'll want to not just expose LevelFilter but actually the whole EnvFilter". So given that EnvFilter is capable of acting like a LevelFilter too I think you're right that is best to just switch to using an EnvFilter.